### PR TITLE
feat: add support for Kubernetes v1.21.0-alpha.1

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

See https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md#changelog-since-v1200

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

**Requirements**:

- [x] Kubernetes artifacts built and pushed by Azure Pipelines
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
Cluster-autoscaler should be bumped to v1.20.0 for this and the previous minor release. I'll do that in a separate PR.

### CODE REVIEW

File: .pipelines/pr-e2e.yaml [hunk 1]
```diff
@@ -89,6 +89,16 @@ jobs:
...
+  parameters:
+    name: 'k8s_1_21_docker_e2e'
+    k8sRelease: '1.21'
+    apimodel: 'examples/e2e-tests/kubernetes/release/default/definition.json'
+    createVNET: true
+    enableKMSEncryption: false
+    containerRuntime: 'docker'
+    runSSHTests: true
+
+    - template: e2e-job-template.yaml
```
Comment (jackfrancis) [inline .pipelines/pr-e2e.yaml:101]: I think this extra indentation is too cute for yaml

Comment (mboersma) [reply .pipelines/pr-e2e.yaml]: D'oh, thanks for catching that! 👍🏼

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by jackfrancis** [comment]
Let's keep the 1.21.0 test config additions

**Pull request merged**